src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -169,6 +186,72 @@ private Section createNewSection(Section requestSection, Section modifiedSection
         throw new IllegalArgumentException();
     }
 
+    private void removeSection(Station deletedStation, List<Section> sectionsContainDeletedStation) {
+        if (sectionsContainDeletedStation.size() == MATCH_ONLY_ONE_SECTION) {
+            removeEdgeSection(sectionsContainDeletedStation.get(0));
+            return;
+        }
+
+        if (sectionsContainDeletedStation.size() == MATCH_DOUBLE_SECTION) {
+            removeInnerSection(deletedStation, sectionsContainDeletedStation);
+            return;
+        }
+
+        throw new IllegalArgumentException();
+    }
+
+    private void removeInnerSection(Station deletedStation, List<Section> sectionsContainDeletedStation) {
+        Section upSection = findUpSectionByStation(deletedStation, sectionsContainDeletedStation);
+        Section downSection = findDownSectionByStation(deletedStation, sectionsContainDeletedStation);
+
+        sections.remove(upSection);
+        sections.remove(downSection);
+        sections.add(createNewSectionFromRemoveSection(upSection, downSection));
+    }
+
+    private Section findUpSectionByStation(Station deletedStation, List<Section> sectionsContainDeletedStation) {
+        return sectionsContainDeletedStation.stream()
+                .filter(section -> section.isEqualDownStation(deletedStation))
+                .findFirst()
+                .orElseThrow(IllegalArgumentException::new);
+    }
+
+    private Section findDownSectionByStation(Station deletedStation, List<Section> sectionsContainDeletedStation) {
+        return sectionsContainDeletedStation.stream()
+                .filter(section -> section.isEqualUpStation(deletedStation))
+                .findFirst()
+                .orElseThrow(IllegalArgumentException::new);
+    }
+
+    private Section createNewSectionFromRemoveSection(Section upSection, Section downSection) {
+        return Section.of(upSection.getUpStation()
+                , downSection.getDownStation()
+                , upSection.getDistance() + downSection.getDistance());
+    }
+
+    private void removeEdgeSection(Section removedSection) {
+        sections.remove(removedSection);
+    }
+
+    private void validateAbleToDeleteSectionsSize() {
+        if (sections.size() < MIN_SECTIONS_SIZE) {
+            throw new CannotDeleteSectionException(ERROR_MESSAGE_MIN_SECTIONS);
+        }
+    }
+
+    private void validateSectionsContainDeleteStationEmpty(List<Section> sectionsContainDeletedStation) {
```

Comment:
메서드명이 너무 길고 무슨 의미인지 파악이 어렵습니다.
이 메서드는 파라미터로 전달 받은 list가 비었으면 예외를 던지는 기능만 합니다.
삭제할 역이 포함된 sections를 찾아와서 비어있는지 확인하는 메서드가 아닙니다.
메서드는 한 가지 기능만 하도록 만들었으면 적절한 이름을 붙여주세요.

```suggestion
    private void validateNonEmpty(List<Section> sections) {
```